mono/utils/mono-error-internals.h

```diff
@@ -43,10 +43,34 @@ struct _MonoErrorBoxed {
 	MonoImage *image;
 };
 
-#define error_init(error) do {	\
-	((MonoErrorInternal*)(error))->error_code = MONO_ERROR_NONE;	\
-	((MonoErrorInternal*)(error))->flags = 0;	\
-} while (0);
+// FIXME This is duplicated.
+#define ERROR_DECL \
+	MonoError error_value; \
+	MonoError * const error = &error_value; \
+	error_init_internal (error);
+
+#define ERROR_DECL_NAME(x) \
+	MonoError x; \
+	error_init_internal (&x);
+
+// MonoError.hidden_0 is MonoErrorInternal.flags.
+// error_init is only now for compatibility at API boundaries.
+// Internal users should use ERROR_DECL.
+
+#undef error_init
```

Comment:
why `#undef`?

Reply:
Because it might be defined already.

Reply:
if `eror_init` is being defined twice, that's a bug and the build should break.

Reply:
Not if it is defined to be the same thing..which it can be if I remove the first two lines. Anyway, let me see about putting this all in just once.